src/scanoss/file_filters.py

```diff
@@ -447,24 +500,26 @@ def _should_skip_file(self, file_rel_path: str, operation_type: str) -> bool:
         if not self.hidden_files_folders and file_name.startswith('.'):
             self.print_debug(f'Skipping file: {file_rel_path} (hidden file)')
             return True
-
         if self.all_extensions:
             return False
-
-        if file_name.lower() in DEFAULT_SKIPPED_FILES:
+        file_name_lower = file_name.lower()
+        # Look for exact files
+        if file_name_lower in DEFAULT_SKIPPED_FILES:
             self.print_debug(f'Skipping file: {file_rel_path} (matches default skip file)')
             return True
-
+        # Look for file endings
         for ending in DEFAULT_SKIPPED_EXT:
-            if file_name.lower().endswith(ending):
+            if file_name_lower.endswith(ending):
                 self.print_debug(f'Skipping file: {file_rel_path} (matches default skip ending: {ending})')
                 return True
-
-        patterns = self._get_operation_patterns(operation_type)
-        if patterns:
-            spec = GitIgnoreSpec.from_lines(patterns)
-            if spec.match_file(file_rel_path):
-                self.print_debug(f'Skipping file: {file_rel_path} (matches custom pattern)')
-                return True
-
+        # Look for custom (extra) endings
+        if self.skip_extensions:
```

Comment:
maybe is cleaner to write it just in one line like 

`if self.skip_extensions and any(file_name_lower.endswith(ending) for ending in self.skip_extensions)` or something like that